comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go

```diff
@@ -284,6 +281,13 @@ func (ia *inventoryagent) fetchSecurityAgentMetadata() {
 	ia.data["feature_cspm_host_benchmarks_enabled"] = securityCfg.GetBool("compliance_config.host_benchmarks.enabled")
 }
 
+func (ia *inventoryagent) fetchTraceAgentMetadata() {
+	traceCfg := ia.getCorrectConfig("trace-agent", fetchTraceConfig)
+
+	ia.data["config_apm_dd_url"] = clean(traceCfg.GetString("apm_config.apm_dd_url"))
+	ia.data["feature_apm_enabled"] = traceCfg.GetBool("apm_config.enabled")
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ Question:** ‏Can we not send these if the call to the trace agent doesn't work? I feel like NULL in REDAPL tables will be better than wrong data if it ever fails.

I wonder if we can cache these to re-use the older data in case of failure too

Reply:
FWIW [REDAPL supports having NULL in boolean fields](https://beagle.us1.staging.dog/?q=c2VsZWN0ICogZnJvbSBkYXRhZG9nX2FnZW50IHdoZXJlIGZlYXR1cmVfcmVtb3RlX2NvbmZpZ3VyYXRpb25fZW5hYmxlZCBpcyBudWxs&orgid=2&interval=auto&start=2024-02-12+15%3A21%3A42&end=2024-02-12+16%3A21%3A42&format=DDSQL)

My suggestion is not something we'd do backend side, but rather in the agent: if we fail to query the trace agent for payload N, we use data we cached from a successful query N-1, hoping that the trace agent config hasn't changed since. This avoids having empty fields or data that flaps.